### PR TITLE
box: fix several gh-10088 corner cases

### LINKED ISSUE #10397: Replica crashes in a test `gh_10088_apply_deletion_of_replica_from_cluster_on_deleted_replica`
The test crashed in this job: https://github.com/tarantool/tarantool/actions/runs/10318608783/job/28565479328

Here's the backtrace:
```
[013] to_be_deleted | 2024-08-09 11:54:31.820 [56215] main I> removed replica 6ed07c54-ed85-4bc1-ad56-79086e437014
[013] to_be_deleted | 2024-08-09 11:54:31.820 [56215] main I> update replication_synchro_quorum = 2
[013] to_be_deleted | 2024-08-09 11:54:31.820 [56215] main/105/gc I> wal/engine cleanup is resumed
[013] to_be_deleted | Segmentation fault
[013] to_be_deleted |   code: SEGV_MAPERR
[013] to_be_deleted |   addr: 0x8
[013] to_be_deleted |   context: 0x53e43c0
[013] to_be_deleted |   siginfo: 0x53e44f0
[013] to_be_deleted |   rax      0x0                0
[013] to_be_deleted |   rbx      0x54547f0          88426480
[013] to_be_deleted |   rcx      0x54395a0          88315296
[013] to_be_deleted |   rdx      0x5457488          88437896
[013] to_be_deleted |   rsi      0x7f541d380740     139999244191552
[013] to_be_deleted |   rdi      0x0                0
[013] to_be_deleted |   rsp      0x7f541d3808f0     139999244191984
[013] to_be_deleted |   rbp      0x7f53fc001038     139998686875704
[013] to_be_deleted |   r8       0x54395b0          88315312
[013] to_be_deleted |   r9       0x5439460          88314976
[013] to_be_deleted |   r10      0x7f541d580b70     139999246289776
[013] to_be_deleted |   r11      0x14               20
[013] to_be_deleted |   r12      0x7f53fc001028     139998686875688
[013] to_be_deleted |   r13      0x7f541d380a20     139999244192288
[013] to_be_deleted |   r14      0x7f53fc001020     139998686875680
[013] to_be_deleted |   r15      0x0                0
[013] to_be_deleted |   rip      0x53d9bf           5495231
[013] to_be_deleted |   eflags   0x10202            66050
[013] to_be_deleted |   cs       0x33               51
[013] to_be_deleted |   gs       0x0                0
[013] to_be_deleted |   fs       0x0                0
[013] to_be_deleted |   cr2      0x8                8
[013] to_be_deleted |   err      0x4                4
[013] to_be_deleted |   oldmask  0x0                0
[013] to_be_deleted |   trapno   0xe                14
[013] to_be_deleted | Current time: 1723204471
[013] to_be_deleted | Please file a bug at https://github.com/tarantool/tarantool/issues
[013] to_be_deleted | Attempting backtrace... Note: since the server has already crashed, 
[013] to_be_deleted | this may fail as well
[013] to_be_deleted | #1  0x613f21 in crash_signal_cb+129
[013] to_be_deleted | #2  0x7f541e98f420 in __restore_rt+0
[013] to_be_deleted | #3  0x53d9bf in diag_raise()+15
[013] to_be_deleted | #4  0x54353e in applier_process_batch(cmsg*)+846
[013] to_be_deleted | #5  0x53e2c5 in applier_subscribe(applier*)+885
[013] to_be_deleted | #6  0x544184 in applier_f(__va_list_tag*)+196
[013] to_be_deleted | #7  0x43057c in fiber_cxx_invoke(int (*)(__va_list_tag*), __va_list_tag*)+12
[013] to_be_deleted | #8  0x61b1b7 in fiber_loop+55
[013] to_be_deleted | #9  0xe6b0f4 in coro_init+68
```

### PR DESCRIPTION
**Pull request by CuriousGeorgiy**
This patch fixes several corner cases of #10088.

Follows up #10088
Closes #10397

### CODE REVIEW

File: src/box/applier.cc [hunk 1]
```diff
@@ -1653,6 +1653,7 @@ applier_apply_tx(struct applier *applier, struct stailq *rows)
 			       &replicaset.applier.order_latch);
```
Comment (Serpentian) [inline src/box/applier.cc:1653]: Please, let's not use links in the commit title. Looks broken and ugly:

image

**Review by Serpentian — APPROVED** [review]
Thank you for the patch! The test is stable for me after it

**Review by sergepetrenko — APPROVED** [review]
Thanks for such a quick fix!

**Comment by sergepetrenko** [comment]
Please reassign me with full-ci once you fix Nikita's comment.

File: src/box/applier.cc (hunk 1 above)
Comment (CuriousGeorgiy) [reply src/box/applier.cc:1653]: Fixed.
